src/resolve/sourceComponent.ts

```diff
@@ -131,8 +133,23 @@ export class SourceComponent implements MetadataComponent {
     return this.markedForDelete;
   }
 
-  public setMarkedForDelete(asDeletion: boolean): void {
-    this.markedForDelete = asDeletion;
+  public getDestructiveChangesType(): DestructiveChangesType {
+    return this.destructiveChangesType;
+  }
+
+  public setMarkedForDelete(destructiveChangeType?: DestructiveChangesType | boolean): void {
+    if (
+      destructiveChangeType === DestructiveChangesType.PRE ||
+      destructiveChangeType === DestructiveChangesType.POST ||
+      destructiveChangeType === true ||
+      destructiveChangeType === undefined
+    ) {
+      this.markedForDelete = true;
+      this.destructiveChangesType =
+        (destructiveChangeType as DestructiveChangesType) || DestructiveChangesType.POST;
```

Comment:
if `true` is passed in, you're making an invalid type assertion `true as DestructiveChangesType`.

What all would the `else` condition handle?  `false` only?  If so, maybe invert the order and return early?   

then you could have this.markedForDelete = true (because it's not false)
and then everything except PRE would mean POST, right?